src/distance_finder.cpp

```diff
@@ -266,6 +292,31 @@ class DistanceFinder {
         double closest_angle;
         getClosestObject(selected_points, closest_distance, closest_angle);
 
+        // get all points on the circle, points that fall within a distance of each other
+        std::vector<lidarPoint> circle_points;
+        circle_points.push_back(lidarPoint(closest_distance, closest_angle));
+        for (int i = 0; i < selected_points.size(); i++) {
+            double i_dist = selected_points[i].getDistance();
+            double i_angle = selected_points[i].getAngle();
+            double dist = sqrt(pow(i_dist, 2) + pow(closest_distance, 2) - 2*i_dist*closest_distance*cos(closest_angle - i_angle));
+            if (dist < lidar_point_range_) {
+                circle_points.push_back(selected_points[i]);
+            }
+        }
+        if (circle_points.size()<min_circle_pts_) {
+            ROS_WARN_STREAM(TAG << "Expected at least " << min_circle_pts_ << " points for radius calculation but got " << circle_points.size());
+            return;
+        }
+
+        // do radius checking, and confirm that the prop found has the expected radius
+        lidarCalculations lidarCalc;
+        double radius = lidarCalc.calculateRadius(circle_points, min_circle_pts_);
+        if (!((radius > marker_radius_ - prop_range_  && radius < marker_radius_ + prop_range_ ) || (radius > sm_buoy_radius_ - prop_range_ && radius < sm_buoy_radius_ + prop_range_) || (radius > lg_buoy_radius_ - prop_range_ && radius < lg_buoy_radius_ + prop_range_))) {
+            // if the prop doesn't fit a radius range, then it doesn't have a valid radius
+            // to be more specific, get label for prop then specify what radius to compare to based on that
```

Comment:
Can you make a separate asana task for this? Making sure the radius is correct based on the prop's label?